maestro/trainer/models/florence_2/core.py

```diff
@@ -0,0 +1,356 @@
+        device (torch.device): Device to use for training.
+        cache_dir (Optional[str]): Directory to cache the model.
+        epochs (int): Number of training epochs.
+        optimizer (Literal["sgd", "adamw", "adam"]): Optimizer to use for training.
+        lr (float): Learning rate for the optimizer.
+        lr_scheduler (Literal["linear", "cosine", "polynomial"]): Learning rate
+            scheduler.
+        batch_size (int): Batch size for training.
+        val_batch_size (Optional[int]): Batch size for validation.
+        num_workers (int): Number of workers for data loading.
+        val_num_workers (Optional[int]): Number of workers for validation data loading.
+        lora_r (int): Rank of the LoRA update matrices.
+        lora_alpha (int): Scaling factor for the LoRA update.
+        lora_dropout (float): Dropout probability for LoRA layers.
+        bias (Literal["none", "all", "lora_only"]): Which bias to train.
+        use_rslora (bool): Whether to use RSLoRA.
+        init_lora_weights (Union[bool, LoraInitLiteral]): How to initialize LoRA
+            weights.
+        output_dir (str): Directory to save output files.
+        metrics (List[BaseMetric]): List of metrics to track during training.
+    """
+    dataset: str
+    model_id: str = DEFAULT_FLORENCE2_MODEL_ID
+    revision: str = DEFAULT_FLORENCE2_MODEL_REVISION
+    device: torch.device = DEVICE
+    cache_dir: Optional[str] = None
+    epochs: int = 10
+    optimizer: Literal["sgd", "adamw", "adam"] = "adamw"
+    lr: float = 1e-5
+    lr_scheduler: Literal["linear", "cosine", "polynomial"] = "linear"
+    batch_size: int = 4
+    val_batch_size: Optional[int] = None
+    num_workers: int = 0
+    val_num_workers: Optional[int] = None
+    lora_r: int = 8
+    lora_alpha: int = 8
+    lora_dropout: float = 0.05
+    bias: Literal["none", "all", "lora_only"] = "none"
+    use_rslora: bool = True
+    init_lora_weights: Union[bool, LoraInitLiteral] = "gaussian"
+    output_dir: str = "./training/florence-2"
+    metrics: List[BaseMetric] = field(default_factory=list)
+
+
+def train(config: TrainingConfiguration) -> None:
+    make_it_reproducible(avoid_non_deterministic_algorithms=False)
+    run_dir = create_new_run_directory(
+        base_output_dir=config.output_dir,
+    )
+    config = replace(
+        config,
+        output_dir=run_dir,
+    )
+    checkpoint_manager = CheckpointManager(run_dir)
+    
+    processor, model = load_model(
+        model_id_or_path=config.model_id,
+        revision=config.revision,
+        device=config.device,
+        cache_dir=config.cache_dir,
+    )
+    train_loader, val_loader, test_loader = prepare_data_loaders(
+        dataset_location=config.dataset,
+        train_batch_size=config.batch_size,
+        processor=processor,
+        device=config.device,
+        num_workers=config.num_workers,
+        test_loaders_workers=config.val_num_workers,
+    )
+    peft_model = prepare_peft_model(
```

Comment:
approaching with peft is also good way to start. FYI, I have tried three different technique
1. Full finetuning
2. Part finetuning (freezing encoder-like part)
3. peft
it turns out 2, 3 is robust for other hyperparameter option and I couldn't fine any stable configuration for 1.

Reply:
I think for now supporting only peft would be enough (which means not the highest priority), since this is the moment of just starting to make the growth.

Reply:
I think so too! For the time being, we have to remember that such an option may arise at some point.